### DISCUSSION TITLE
How to start to modify the source?

### DISCUSSION THREAD

**Discussion by nvlong248** [SCADA-LTS/Scada-LTS#3013] [General]
Hi,
I am new at open source software.
I have been using SCADA-LTS for a while, but have no idea about starting modification it.
Could you give me some tips and documents to start modifying it?
Thank you!

**Comment by rafaelpsousa** [comment]
start with
https://github.com/SCADA-LTS/Scada-LTS/wiki/Prepare-environment-to-develop-(IntelliJ-Community)

I use Eclipse IDE to make changes, and gradle to compile

**Comment by Limraj** [comment]
Hi @nvlong248, @rafaelpsousa, with details:
https://github.com/SCADA-LTS/Scada-LTS/wiki/Getting-started-Development-%E2%80%90-Linux
https://github.com/SCADA-LTS/Scada-LTS/wiki/Getting-started-Development-%E2%80%90-Windows

We recommend using gradle tasks(from Intellij), which contain all the necessary steps from compiling, running tests, to creating a war and deploying the application to tomcat, and starting tomcat itself:
https://github.com/SCADA-LTS/Scada-LTS/wiki/Getting-started-Development-%E2%80%90-Linux#51-gradle-tasks-
https://github.com/SCADA-LTS/Scada-LTS/wiki/Getting-started-Development-%E2%80%90-Windows#51-gradle-tasks-

**Comment by Limraj** [comment]
And @nvlong248, @rafaelpsousa,
we encourage you to report issues, and propose your own fixes to issues. 

1. It is best to look at our approach to naming branches, creating messages for commits. 
- Pattern branch name: 
fix_or_feature/#issue_number__issue_title 
- Pattern message in commits: 
#issue_number - issue_title - your_description; 

We would appreciate it if you followed these naming rules. 

3. When we work on a version like 2.8.0, as we do now, we create a dedicated branch release/2.8.0 for it, to which we collect all changes and fixes, and at release we merge with the develop branch.

4. To avoid configuration discrepancies and potential problems with project operation, we recommend using Intellij, the Community Edition is sufficient.

Best regards.
